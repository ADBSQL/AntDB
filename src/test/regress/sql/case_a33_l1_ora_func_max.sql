set grammar to oracle;
CREATE TABLE t4test (id int,name varchar,salary int,dep varchar);
insert into t4test values(1,'xu',12000,'a1');
insert into t4test values(2,'zhang',12500,'a2');
insert into t4test values(3,'li',13500,'a1');
insert into t4test values(4,'lin',13500,'a3');
insert into t4test values(5,'xu',23456,'a1');
insert into t4test values(6,'zha0',12500,'a1');
insert into t4test values(7,'xiao',13500,'a2');
insert into t4test values(8,'yang',23500,'a3');

SELECT max(salary) from t4test;

SELECT dep, name, salary,
	MAX(salary) OVER (PARTITION BY dep) AS dep_max
	FROM t4test
	ORDER BY dep, name, salary;
	
	
SELECT dep, name, salary
	FROM (SELECT dep, name, salary, 
			MAX(salary) OVER (PARTITION BY dep) AS max_sal
			FROM t4test)
	WHERE salary = max_sal
	ORDER BY dep, name, salary;	
	

drop table t4test;