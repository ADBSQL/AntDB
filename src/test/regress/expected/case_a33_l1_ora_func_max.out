set grammar to oracle;
CREATE TABLE t4test (id int,name varchar,salary int,dep varchar);
insert into t4test values(1,'xu',12000,'a1');
insert into t4test values(2,'zhang',12500,'a2');
insert into t4test values(3,'li',13500,'a1');
insert into t4test values(4,'lin',13500,'a3');
insert into t4test values(5,'xu',23456,'a1');
insert into t4test values(6,'zha0',12500,'a1');
insert into t4test values(7,'xiao',13500,'a2');
insert into t4test values(8,'yang',23500,'a3');
SELECT max(salary) from t4test;
 MAX(SALARY)  
--------------
        23500
(1 row)

SELECT dep, name, salary,
	MAX(salary) OVER (PARTITION BY dep) AS dep_max
	FROM t4test
	ORDER BY dep, name, salary;
 DEP | NAME  | SALARY | DEP_MAX 
-----+-------+--------+---------
 a1  | li    |  13500 |   23456
 a1  | xu    |  12000 |   23456
 a1  | xu    |  23456 |   23456
 a1  | zha0  |  12500 |   23456
 a2  | xiao  |  13500 |   13500
 a2  | zhang |  12500 |   13500
 a3  | lin   |  13500 |   23500
 a3  | yang  |  23500 |   23500
(8 rows)

	
	
SELECT dep, name, salary
	FROM (SELECT dep, name, salary, 
			MAX(salary) OVER (PARTITION BY dep) AS max_sal
			FROM t4test)
	WHERE salary = max_sal
	ORDER BY dep, name, salary;	
 DEP | NAME | SALARY 
-----+------+--------
 a1  | xu   |  23456
 a2  | xiao |  13500
 a3  | yang |  23500
(3 rows)

	
drop table t4test;
